modules/secrets-store-vault.adoc

```diff
@@ -0,0 +1,384 @@
+// Module included in the following assemblies:
+//
+// * nodes/pods/nodes-pods-secrets-store.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="secrets-store-vault_{context}"]
+= Mounting secrets from HashiCorp Vault
+
+You can use the {secrets-store-operator} to mount secrets from HashiCorp Vault to a CSI volume in {product-title}.
+
+// TODO: Is there a requirement for particular provider? https://issues.redhat.com/browse/OSDOCS-10136 steps say Azure, is this true?
+// To mount secrets from HashiCorp Vault, your cluster must be installed on Microsoft Azure.
+
+.Prerequisites
+
+// TODO
+// * Your cluster is installed on Azure.
+* You have installed the {secrets-store-operator}. See _Installing the {secrets-store-driver}_ for instructions.
+* You have installed Helm.
+* You have installed HashiCorp Vault (`vault`).
+* You have configured HashiCorp Vault to store the required secrets.
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Add the HashiCorp Helm repository by running the following command:
++
+[source,terminal]
+----
+$ helm repo add hashicorp https://helm.releases.hashicorp.com
+----
+
+. Update all repositories to ensure that Helm is aware of the latest versions by running the following command:
++
+[source,terminal]
+----
+$ helm repo update
+----
+
+. Create a new project for Vault by running the following command:
++
+[source,terminal]
+----
+$ oc new-project vault
```

Comment:
@dobsonj Following the steps from @ropatil010, this is using a new `vault` namespace. The AWS/Azure procedures use the `openshift-cluster-csi-drivers` namespace throughout. Do we need to update to use `openshift-cluster-csi-drivers` too, or can we leave as `vault`?

Reply:
From my end, better to use with vault esle if we are using ns openshift-cluster-csi-drivers ns then we need to use this value in many places instead of vault.

Reply:
Okay thanks, I'll leave as is as long as @dobsonj doesn't have any objections.

Reply:
I have no objection. Vault is a little different from the other providers, in that you need a vault server (if you don't already have one deployed in your environment), as opposed to using a service offered by the cloud provider. It's better to deploy the server in its own namespace.

Reply:
Great thanks, I'll leave it as is!